File: ios/Runner/SensorHandler.swift
import Foundation

/// Sensor reads feeding the ADR-003 fitness score.
///
/// Phase 0 stub: real battery / thermal (ProcessInfo.thermalState) / RSSI reads
/// land on-device.
final class SensorHandler: SensorApi {
  func getBatteryLevel(completion: @escaping (Result<Int64, Error>) -> Void) {
    completion(.failure(NotImplemented.error("SensorApi.getBatteryLevel")))
  }

  func getThermalState(completion: @escaping (Result<ThermalState, Error>) -> Void) {
    completion(.failure(NotImplemented.error("SensorApi.getThermalState")))
  }

  func getRssiToPeers(completion: @escaping (Result<[String: Int64], Error>) -> Void) {
    completion(.failure(NotImplemented.error("SensorApi.getRssiToPeers")))
  }
}

File: android/app/src/main/kotlin/com/fzrilsh/toice/SensorHandler.kt
package com.fzrilsh.toice

/**
 * Sensor reads feeding the ADR-003 fitness score.
 *
 * Phase 0 stub: real battery / thermal / RSSI reads land on-device.
 */
class SensorHandler : SensorApi {
  override fun getBatteryLevel(callback: (Result<Long>) -> Unit) =
    callback(notImplemented("SensorApi.getBatteryLevel"))

  override fun getThermalState(callback: (Result<ThermalState>) -> Unit) =
    callback(notImplemented("SensorApi.getThermalState"))

  override fun getRssiToPeers(callback: (Result<Map<String, Long>>) -> Unit) =
    callback(notImplemented("SensorApi.getRssiToPeers"))
}
